transport/socks5/socks5.go

```diff
@@ -11,6 +11,17 @@ import (
 	"strconv"
 )
 
+// AuthMethod is the authentication method as defined in RFC 1928 section 3.
+type AuthMethod = uint8
+
+// SOCKS authentication methods as defined in RFC 1928 section 3.
+const (
+	MethodNoAuth       AuthMethod = 0x00
```

Comment:
Yeah, I mean we can use iota here like this:
```go
const (
	MethodNoAuth uint8 = iota
	MethodGSSAPI
	MethodUserPass
	MethodNoAcceptable = 0xFF
)
```

or just define what we need like this:

```go
const (
	MethodNoAuth = 0x00
	MethodUserPass = 0x02
)
```

Reply:
I would prefer specifying them out as written on the RFC, it's more explicit for viewers.

Reply:
I deleted the ones we don't need